pandas/tests/copy_view/test_indexing.py

```diff
@@ -11,13 +13,47 @@
 import pandas._testing as tm
 from pandas.tests.copy_view.util import get_array
 
+
+@pytest.fixture(params=["numpy", "nullable"])
+def backend(request):
+    if request.param == "numpy":
+
+        def make_dataframe(*args, **kwargs):
+            return DataFrame(*args, **kwargs)
+
+        def make_series(*args, **kwargs):
+            return Series(*args, **kwargs)
+
+    elif request.param == "nullable":
+
+        def make_dataframe(*args, **kwargs):
+            df = DataFrame(*args, **kwargs)
+            df_nullable = df.convert_dtypes()
+            # convert_dtypes will try to cast float to int if there is no loss in
+            # precision -> undo that change
+            for col in df.columns:
+                if is_float_dtype(df[col].dtype) and not is_float_dtype(
+                    df_nullable[col].dtype
+                ):
+                    df_nullable[col] = df_nullable[col].astype("Float64")
+            # copy final result to ensure we start with a fully self-owning DataFrame
+            return df_nullable.copy()
+
+        def make_series(*args, **kwargs):
+            ser = Series(*args, **kwargs)
+            return ser.convert_dtypes().copy()
+
+    return request.param, make_dataframe, make_series
```

Comment:
I experimented locally with some different ways to parametrize this, and settled on the above with a fixture that returns adapted constructor functions. Alternative was some "postprocess" function that was called on each dataframe/series that was created. Actually parametrizing on dtype objects seems more complicated.

Side note: inside the test I am assigning those factory functions to `DataFrame` and `Series` to avoid more changes to the test code, but that can also be a different name (like the `make_dataframe`/`make_series` used above) to be more explicit. The reason I currently preferred this somewhat magic override, is that this gives a smaller diff, and this keeps the test code more copy-pastable (and I think the idea is that we have an option that controls the behaviour of the constructors in the future, so that would give less changes then).

Reply:
This is easier to follow than I expected, so looks good generally